### ISSUE TITLE
`baseFeePerGas` field in ExecutionPayloadV1 is too permissive 

### ISSUE THREAD

**Issue by marioevz** [sigp/lighthouse#3314]
## Description

Execution client's Engine API specifies `baseFeePerGas` as a `QUANTITY` field which has the following restrictions:

> Values of a field of QUANTITY type MUST be encoded as a hexadecimal string with a 0x prefix and the leading 0s stripped (except for the case of encoding the value 0) matching the regular expression ^0x(?:0|(?:[a-fA-F1-9][a-fA-F0-9]*))$.
> 
> Note: Byte order of encoded value having QUANTITY type is big-endian.

Currently is possible to feed lighthouse the following spoofed values and they will end up in the Execution payload without issues:
- `Spoofing (node X) baseFeePerGas, 0xce187 -> ce187` (No `0x` prefix)
- `Spoofing (node X) baseFeePerGas, 0xeb89a -> 0x0eb89a` (Leading zero)

## Version

Lighthouse/v2.3.2-rc.0-d5e2d98

## Present Behaviour

The non-compliant values are accepted when returned in the `ExecutionPayloadV1`.

## Expected Behaviour

Lighthouse should reject the payload because of the incorrect formatting of this value.

## Reproduce Steps

Using hive branch https://github.com/marioevz/hive/tree/all-tests-plus-cl-clients, run:
```
./hive --client go-ethereum,lighthouse-bn,lighthouse-vc --sim eth2/engine --sim.loglevel 5
```

**Comment by marioevz** [comment]
I've updated the hive simulator branch, please update to the latest commit and use this command instead to run only a single test case:
```
./hive --client go-ethereum,lighthouse-bn,lighthouse-vc --sim eth2/engine --sim.loglevel 5 --sim.limit "/invalid-quantity-fields"
```

**Comment by michaelsproul** [comment]
Fixed by #3321
